src/Cake.Issues.Testing/IssueChecker.cs

```diff
@@ -266,6 +270,12 @@ public static class IssueChecker
                 throw new Exception(
                     $"Expected issue.RuleUrl to be '{ruleUrl?.ToString()}' but was '{issue.RuleUrl?.ToString()}'.");
             }
+
+            if (!issue.AdditionalInformation.ContentEquals(additionalInformation))
+            {
+                throw new Exception(
+                    $"Expected issue.AdditionalInformation to be '{additionalInformation}' but was '{issue.AdditionalInformation}'.");
```

Comment:
This will have the type in the exception message like this:

```
'Expected issue.AdditionalInformation to be 'System.Collections.Generic.Dictionary`2[System.String,System.String]' but was 'System.Collections.Generic.Dictionary`2[System.String,System.String]'.'
```

We should return a helpful message.

Reply:
@pascalberger Yeeeeah, I did see that too... But I didnt know what to do there, I didn't want to do a oversized algo there for "testing purposes", so what do you have in mind to print there that is better from a consumer perspective?

Reply:
@janniksam See comment above: I would suggest to implement the comparison directly instead of using an extension. To improve error message we can do something like: 

1. Check if all expected keys exist and match the value. If not print out the name of the missing key or not matching value 
2. Check if there exist keys which are not expected and print out their name